tests/components/bosch_alarm/test_services.py

```diff
@@ -0,0 +1,153 @@
+"""Tests for Bosch Alarm component."""
+
+import asyncio
+from collections.abc import AsyncGenerator
+from unittest.mock import AsyncMock, patch
+
+import pytest
+import voluptuous as vol
+
+from homeassistant.components.bosch_alarm.const import (
+    ATTR_CONFIG_ENTRY_ID,
+    ATTR_DATETIME,
+    DOMAIN,
+    SERVICE_SET_DATE_TIME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError, ServiceValidationError
+from homeassistant.util import dt as dt_util
+
+from . import setup_integration
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture(autouse=True)
+async def platforms() -> AsyncGenerator[None]:
+    """Return the platforms to be loaded for this test."""
+    with patch("homeassistant.components.bosch_alarm.PLATFORMS", []):
+        yield
+
+
+async def test_set_date_time_service(
+    hass: HomeAssistant,
+    mock_panel: AsyncMock,
+    area: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test that the service calls succeed if the service call is valid."""
+    await setup_integration(hass, mock_config_entry)
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SET_DATE_TIME,
+        {
+            ATTR_CONFIG_ENTRY_ID: mock_config_entry.entry_id,
+            ATTR_DATETIME: dt_util.now(),
+        },
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    mock_panel.set_panel_date.assert_called_once()
+
+
+async def test_set_date_time_service_fails_bad_entity(
+    hass: HomeAssistant,
+    mock_panel: AsyncMock,
+    area: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test that the service calls fail if the service call is done for an incorrect entity."""
+    await setup_integration(hass, mock_config_entry)
+    with pytest.raises(ServiceValidationError, match=".."):
```

Comment:
with `..` I meant, fill in the error that the user sees, let's also do this for the other raises